### DISCUSSION TITLE
Is it possible to map a Prisma schema to an object?

### DISCUSSION THREAD

**Discussion by serg06** [prisma/prisma#17093] [Q&A] [answered]
Hello, I have a table like this:

```
model User {
  id                     Int
  name               String
  video_settingsMax_bitrate Int
  video_settingsMax_filesize Int
  audio_settingsMax_bitrate Int
  audio_settingsMax_filesize Int
  image_settingsMax_bitrate Int
  image_settingsMax_filesize Int
}
```

I'd like its TypeScript type to be this:

```ts
interface User {
  id: number
  name: string
  video_settings: {
    max_bitrate: number
    max_filesize: number
  }
  audio_settings: {
    max_bitrate: number
    max_filesize: number
  }
  image_settings: {
    max_bitrate: number
    max_filesize: number
  }
}
```

Is it possible to create that mapping, or do I need a separate model & relation?

**Comment by nurul3101** [comment, marked as answer]
Hello @serg06 👋 

Prisma would generate type based on your model format. So in this case, the User type generated by Prisma would look like this:

```typescript
/**
 * Model User
 * 
 */
export type User = {
  id: number
  name: string
  video_settingsMax_bitrate: number
  video_settingsMax_filesize: number
  audio_settingsMax_bitrate: number
  audio_settingsMax_filesize: number
  image_settingsMax_bitrate: number
  image_settingsMax_filesize: number
}
```
which you can import like this:
```typescript
import { PrismaClient, User } from '@prisma/client';
```

To create an interface in the format that you have specified, you would need to define that manually, prisma won't automatically create it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
